From 76b1097443fa01bd454538506489add28db6d273 Mon Sep 17 00:00:00 2001
From: Pratham Malaviya <prathamm@mechasystems.com>
Date: Tue, 13 Jan 2026 16:19:17 +0530
Subject: [PATCH 6/9] fix: display: avoid overwriting init sequence

This patch fixes the issue where it was overwriting
pre-programmed display init sequence. Resulting in
warm colours and other visual changes.

Signed-off-by: Pratham Malaviya <prathamm@mechasystems.com>

diff --git a/drivers/gpu/drm/panel/panel-ddic-ch13726a.c b/drivers/gpu/drm/panel/panel-ddic-ch13726a.c
index fe2413265c2d..6fde61cedf37 100644
--- a/drivers/gpu/drm/panel/panel-ddic-ch13726a.c
+++ b/drivers/gpu/drm/panel/panel-ddic-ch13726a.c
@@ -54,112 +54,13 @@ static int ch13726a_on(struct ch13726a_panel *ctx)
 
 	dsi->mode_flags |= MIPI_DSI_MODE_LPM;
 	
-	mipi_dsi_dcs_write_seq(dsi, 0x2a, 0x00, 0x00, 0x04, 0x37);
-	mipi_dsi_dcs_write_seq(dsi, 0x2b, 0x00, 0x00, 0x04, 0xd7);
-	mipi_dsi_dcs_write_seq(dsi, 0x8f, 0x04, 0x88, 0x77);
-	mipi_dsi_dcs_write_seq(dsi, 0x51, 0xff);
-	mipi_dsi_dcs_write_seq(dsi, 0x53, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0x35, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0x60, 0x80);
-	mipi_dsi_dcs_write_seq(dsi, 0x44, 0x02, 0x6c);
-	mipi_dsi_dcs_write_seq(dsi, 0xf0, 0x50);
-	mipi_dsi_dcs_write_seq(dsi, 0xb0, 0x0e, 0x0e);
-	mipi_dsi_dcs_write_seq(dsi, 0xb2, 0x80);
-	mipi_dsi_dcs_write_seq(dsi, 0xb3, 0x00, 0x1e);
-	mipi_dsi_dcs_write_seq(dsi, 0xb4, 0x92, 0x18, 0x92);
-	mipi_dsi_dcs_write_seq(dsi, 0xb6, 0x00, 0x3f, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xb7, 0x00, 0x00, 0x00, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xb8, 0x02, 0x62);
-	mipi_dsi_dcs_write_seq(dsi, 0xb9, 0x11);
-	mipi_dsi_dcs_write_seq(dsi, 0xbb, 0x08, 0x08);
-	mipi_dsi_dcs_write_seq(dsi, 0xc0, 0x80);
-	mipi_dsi_dcs_write_seq(dsi, 0xc7, 0x40);
-	mipi_dsi_dcs_write_seq(dsi, 0xc9, 0x55, 0xf8, 0xf8);
-	mipi_dsi_dcs_write_seq(dsi, 0xe8, 0x58);
-	mipi_dsi_dcs_write_seq(dsi, 0xec, 0x08);
-	mipi_dsi_dcs_write_seq(dsi, 0xf0, 0x51);
-	mipi_dsi_dcs_write_seq(dsi, 0xb0, 0x28);
-	mipi_dsi_dcs_write_seq(dsi, 0xb1, 0x28);
-	mipi_dsi_dcs_write_seq(dsi, 0xb2, 0x1c);
-	mipi_dsi_dcs_write_seq(dsi, 0xb3, 0x1c);
-	mipi_dsi_dcs_write_seq(dsi, 0xc0, 0x41);
-	mipi_dsi_dcs_write_seq(dsi, 0xc1, 0x1e, 0x1e);
-	mipi_dsi_dcs_write_seq(dsi, 0xc4, 0x28);
-	mipi_dsi_dcs_write_seq(dsi, 0xc9, 0x1b);
-	mipi_dsi_dcs_write_seq(dsi, 0xd1, 0x4a, 0x4a, 0x80, 0x4f, 0x69);
-	mipi_dsi_dcs_write_seq(dsi, 0xd2, 0x05);
-	mipi_dsi_dcs_write_seq(dsi, 0xd3, 0x11, 0x00, 0x00, 0x64);
-	mipi_dsi_dcs_write_seq(dsi, 0xd7, 0xa8, 0x5a, 0xaa, 0xaa);
-	mipi_dsi_dcs_write_seq(dsi, 0xd8, 0xa8, 0x5a, 0xaa, 0xaa);
-	mipi_dsi_dcs_write_seq(dsi, 0xdd, 0xaa, 0xaa, 0xaa, 0xaa);
-	mipi_dsi_dcs_write_seq(dsi, 0xf0, 0x52);
-	mipi_dsi_dcs_write_seq(dsi, 0xb0, 0x21, 0x94, 0x69);
-	mipi_dsi_dcs_write_seq(dsi, 0xba, 0xff, 0x35, 0x43, 0x52, 0x61, 0xff, 0x71, 0x79, 0x7d, 0x80, 0x0f, 0x85, 0x87);
-	mipi_dsi_dcs_write_seq(dsi, 0xb9, 0xaa, 0x21, 0x45, 0x67, 0x86, 0xaa, 0x95, 0xa4, 0xc0, 0xde, 0xfe, 0xfa, 0x09, 0x18, 0x26);
-	mipi_dsi_dcs_write_seq(dsi, 0xb8, 0x00, 0x00, 0xda, 0xed, 0xf8, 0x55, 0x29, 0x51, 0x6a, 0x7b, 0x55, 0xa1, 0xc2, 0xdf, 0xf7);
-	mipi_dsi_dcs_write_seq(dsi, 0xb7, 0xfa, 0xec, 0xfa, 0x07, 0x16, 0xff, 0x24, 0x2b, 0x2e, 0x31, 0x0f, 0x36, 0x37);
-	mipi_dsi_dcs_write_seq(dsi, 0xb6, 0xa9, 0xec, 0x0f, 0x2d, 0x4a, 0xaa, 0x58, 0x66, 0x81, 0x9c, 0xaa, 0xb6, 0xc3, 0xd0, 0xde);
-	mipi_dsi_dcs_write_seq(dsi, 0xb5, 0x00, 0x00, 0xd8, 0xe3, 0xea, 0x55, 0x03, 0x22, 0x3b, 0x51, 0x55, 0x76, 0x95, 0xaf, 0xc5);
-	mipi_dsi_dcs_write_seq(dsi, 0xb4, 0xff, 0x14, 0x23, 0x32, 0x41, 0xff, 0x51, 0x59, 0x5d, 0x61, 0x0f, 0x65, 0x67);
-	mipi_dsi_dcs_write_seq(dsi, 0xb3, 0xaa, 0x05, 0x28, 0x48, 0x67, 0xaa, 0x76, 0x84, 0xa0, 0xbc, 0xea, 0xd9, 0xe8, 0xf7, 0x05);
-	mipi_dsi_dcs_write_seq(dsi, 0xb2, 0x00, 0x00, 0xc0, 0xcd, 0xcf, 0x55, 0x15, 0x39, 0x52, 0x69, 0x55, 0x8f, 0xad, 0xc6, 0xdd);
-	mipi_dsi_dcs_write_seq(dsi, 0xf0, 0x53);
-	mipi_dsi_dcs_write_seq(dsi, 0xb0, 0x00, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xb3, 0xa8, 0x10, 0x3c, 0x0a, 0x03, 0x77, 0x77);
-	mipi_dsi_dcs_write_seq(dsi, 0xb4, 0x26, 0x00, 0x00, 0x05, 0x00, 0x77, 0x77);
-	mipi_dsi_dcs_write_seq(dsi, 0xb9, 0x24, 0x90, 0x00, 0x77, 0x17, 0x31, 0x22, 0x90, 0x00, 0x77, 0x17, 0x31);
-	mipi_dsi_dcs_write_seq(dsi, 0xba, 0x20, 0x90, 0x00, 0x77, 0x03, 0x10, 0x21, 0x90, 0x00, 0x77, 0x03, 0x10);
-	mipi_dsi_dcs_write_seq(dsi, 0xbe, 0x20, 0x09, 0x00, 0x00, 0x2c, 0x20, 0x46, 0x00, 0x00, 0x2c);
-	mipi_dsi_dcs_write_seq(dsi, 0xc1, 0x13, 0x12, 0x0b, 0x0a, 0x02, 0x09, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xc3, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x09, 0x03, 0x0a, 0x0b, 0x12, 0x13);
-	mipi_dsi_dcs_write_seq(dsi, 0xf0, 0x55);
-	mipi_dsi_dcs_write_seq(dsi, 0xb1, 0x80, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xb3, 0x7f, 0x7f, 0x00, 0x00, 0x77, 0x87, 0x32, 0xaa, 0x21);
-	mipi_dsi_dcs_write_seq(dsi, 0xb4, 0x2a, 0xa9, 0x2a, 0x21, 0xaa, 0x32, 0x00, 0x87, 0x77);
-	mipi_dsi_dcs_write_seq(dsi, 0xb5, 0x32, 0xaa, 0x21, 0x32, 0xaa, 0x21, 0x77, 0x87, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xb6, 0x00, 0x87, 0x77, 0x87, 0x77, 0x00, 0x21, 0xaa, 0x32);
-	mipi_dsi_dcs_write_seq(dsi, 0xc3, 0x7f, 0x7f, 0x00, 0x00, 0x47, 0x51, 0x32, 0xaa, 0x21);
-	mipi_dsi_dcs_write_seq(dsi, 0xc4, 0x2a, 0xa9, 0x2a, 0x21, 0x66, 0x1e, 0x00, 0x87, 0x77);
-	mipi_dsi_dcs_write_seq(dsi, 0xc5, 0x1e, 0x66, 0x14, 0x32, 0xaa, 0x21, 0x77, 0x87, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xc6, 0x00, 0x51, 0x47, 0x87, 0x77, 0x00, 0x21, 0xaa, 0x32);
-	mipi_dsi_dcs_write_seq(dsi, 0xc7, 0x7f, 0x7f, 0x00, 0x00, 0x77, 0x87, 0x32, 0xaa, 0x21);
-	mipi_dsi_dcs_write_seq(dsi, 0xc8, 0x32, 0xaa, 0x21, 0x32, 0xaa, 0x21, 0x77, 0x87, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xc4, 0x2a, 0xa9, 0x2a, 0x21, 0xaa, 0x32, 0x00, 0x87, 0x77);
-	mipi_dsi_dcs_write_seq(dsi, 0xb6, 0x00, 0x87, 0x77, 0x87, 0x77, 0x00, 0x21, 0xaa, 0x32);
-	mipi_dsi_dcs_write_seq(dsi, 0xf0, 0x58);
-	mipi_dsi_dcs_write_seq(dsi, 0xc0, 0x83, 0xad, 0x52, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xc1, 0xbf, 0x7f, 0x3f, 0x2f, 0x1f, 0x0f, 0x07, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xc2, 0xff);
-	mipi_dsi_dcs_write_seq(dsi, 0xc3, 0x7f, 0x05);
-	mipi_dsi_dcs_write_seq(dsi, 0xc4, 0x09);
-	mipi_dsi_dcs_write_seq(dsi, 0xc5, 0x00, 0x09, 0x09, 0x09, 0x09, 0x00, 0x09, 0x09, 0x09, 0x09);
-	mipi_dsi_dcs_write_seq(dsi, 0xc6, 0xff, 0xff);
-	mipi_dsi_dcs_write_seq(dsi, 0xc7, 0x5a);
-	mipi_dsi_dcs_write_seq(dsi, 0xc9, 0x7f, 0x3f, 0x2f, 0x1f, 0x0f, 0x07, 0x05, 0x04, 0x03, 0x02, 0x01, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xca, 0xff, 0x8f);
-	mipi_dsi_dcs_write_seq(dsi, 0xcb, 0x67, 0x44, 0x34, 0x01, 0x00, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xcc, 0x5a, 0xe9, 0x1f, 0xdc, 0x8b, 0x01, 0x1b, 0xc8, 0xac, 0x9c, 0x00, 0x89, 0x70, 0x52, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0x59, 0x0e);
-	mipi_dsi_dcs_write_seq(dsi, 0xf0, 0x59);
-	mipi_dsi_dcs_write_seq(dsi, 0xb0, 0x13, 0x04, 0xc4, 0x00, 0x3c, 0x3b);
-	mipi_dsi_dcs_write_seq(dsi, 0xb2, 0x07, 0x73, 0xda, 0xfe, 0x13, 0xe8, 0x13, 0xf9, 0x52, 0x3e, 0x90, 0x2f, 0xb0, 0x02, 0x1b, 0x29);
-	mipi_dsi_dcs_write_seq(dsi, 0xb3, 0xf5, 0x12, 0x1e, 0x29, 0xf1, 0x81, 0x02, 0x1e, 0x13, 0x17, 0x1a, 0x1d, 0x1e, 0x0f, 0x00, 0x00);
-	mipi_dsi_dcs_write_seq(dsi, 0xc2, 0xf1, 0xe1, 0xd1, 0xa1, 0x71, 0x31, 0x02, 0x1e, 0x28, 0xf1, 0x91, 0x12, 0x2e, 0xf5, 0x91, 0x02);
-	mipi_dsi_dcs_write_seq(dsi, 0xc3, 0x2b, 0xb0, 0x03, 0xf9, 0x52, 0x3e, 0x91, 0x3f, 0x81, 0x7e, 0x30, 0xa7, 0xed, 0x0f, 0x00, 0x00);
-	
-	ret = mipi_dsi_dcs_exit_sleep_mode(dsi);
-	if (ret < 0) {
-		dev_err(dev, "Failed to exit sleep mode: %d\n", ret);
-		return ret;
-	}
-	
-	msleep(128);
-	
-	ret = mipi_dsi_dcs_set_display_on(dsi);
-	if (ret < 0) {
-		dev_err(dev, "Failed to set display on: %d\n", ret);
-		return ret;
-	}
+	mipi_dsi_dcs_write_seq(dsi,MIPI_DCS_WRITE_CONTROL_DISPLAY, MIPI_DCS_NOP);
+
+	mipi_dsi_dcs_write_seq(dsi, MIPI_DSI_V_SYNC_END, MIPI_DCS_NOP);
+
+	msleep(120);
+	mipi_dsi_dcs_write_seq(dsi, MIPI_DSI_GENERIC_LONG_WRITE, MIPI_DCS_NOP);
+
 	msleep(20);
 
 	return 0;
-- 
2.25.1

